dao/common.go

```diff
@@ -79,6 +77,20 @@ func BulkMessageFromSource(source *m.Source, authentication *m.Authentication) (
 		authentications[i] = authenticationsByResource[i].ToEvent()
 	}
 
+	applicationAuthenticationDao := &ApplicationAuthenticationDaoImpl{TenantID: &source.TenantID}
+	applicationAuthenticationsFromResource, err := applicationAuthenticationDao.ApplicationAuthenticationsByResource(authentication.ResourceType, source.Applications, authenticationsByResource)
+
+	if err != nil {
+		return nil, err
+	}
+
+	applicationAuthentications := make([]interface{}, len(applicationAuthenticationsFromResource))
```

Comment:
What about using `[]*m.ApplicationAuthenticationEvent` here instead of the `[]interface{}`? That way its easier to know at a glance what the slice holds.

Reply:
`.ToEvent()` needs to return `interface{}` to satisfy the `model.Event` interface. Since we're just marshaling and throwing it onto kafka the type is less important.

Reply:
All right, thank you. In this case it doesn't satisfy that `model.Event` interface, though, so I would probably vote for refactoring that in a follow-up PR.